src/main/scala/sbtbuildinfo/ScalaCaseObjectRenderer.scala

```diff
@@ -63,16 +63,20 @@ private[sbtbuildinfo] case class ScalaCaseObjectRenderer(options: Seq[BuildInfoO
   def toJsonLine: Seq[String] =
     if (options contains BuildInfoOption.ToJson)
       List(
-         """  val toJson: String = toMap.map{ i =>
-           |    def quote(x:Any) : String = "\"" + x + "\""
-           |    val key : String = quote(i._1)
-           |    val value : String = i._2 match {
-           |       case elem : Seq[_] => elem.map(quote).mkString("[", ",", "]")
-           |       case elem : Option[_] => elem.map(quote).getOrElse("null")
-           |       case elem => quote(elem)
+         """  private def quote(x: Any): String = "\"" + x + "\""
+           |
+           |  private def toJsonValue(value: Any): String = {
+           |    value match {
+           |      case elem: Seq[_] => elem.map(toJsonValue).mkString("[", ",", "]")
+           |      case elem: Option[_] => elem.map(toJsonValue).orNull
+           |      case elem: Map[String, Any] => elem.map {
```

Comment:
(Minor - feature creep): We could also match on `Map[String, Any]` and sending the key through `toJsonValue` here. IIUC

Reply:
Yes, I think we can do it